skyportal/tests/frontend/test_followup_requests.py

```diff
@@ -185,3 +188,111 @@ def test_delete_followup_request(
     driver.wait_for_xpath_to_disappear(
         '''//table[contains(@data-testid, "followupRequestTable")]//td[contains(., "submitted")]'''
     )
+
+
+@pytest.mark.flaky(reruns=2)
+@pytest.mark.skipif(not sedm_isonline, reason="SEDM server down")
+def test_submit_new_followup_request_two_groups(
+    driver,
+    super_admin_user,
+    public_source_two_groups,
+    super_admin_token,
+    public_group,
+    public_group2,
+    view_only_token_group2,
+    user_group2,
+):
+
+    idata = add_telescope_and_instrument("SEDM", super_admin_token)
+    add_allocation(idata['id'], public_group.id, super_admin_token)
+
+    driver.get(f"/become_user/{super_admin_user.id}")
+
+    driver.get(f"/source/{public_source_two_groups.id}")
+    # wait for the plots to load
+    driver.wait_for_xpath('//div[@class="bk-root"]//span[text()="Flux"]', timeout=20)
+    # this waits for the spectroscopy plot by looking for the element Mg
+    driver.wait_for_xpath('//div[@class="bk-root"]//label[text()="Mg"]', timeout=20)
+
+    submit_button = driver.wait_for_xpath(
+        '//form[@class="rjsf"]//button[@type="submit"]'
+    )
+
+    group_select = driver.wait_for_xpath('//*[@id="selectGroups"]')
+    driver.scroll_to_element_and_click(group_select)
+
+    group1 = driver.wait_for_xpath(f'//*[@data-testid="group_{public_group.id}"]')
+    driver.scroll_to_element_and_click(group1)
+
+    group2 = driver.wait_for_xpath(f'//*[@data-testid="group_{public_group2.id}"]')
+    driver.scroll_to_element_and_click(group2)
+
+    body = driver.wait_for_xpath('//body')
+    driver.scroll_to_element_and_click(body)
+
+    mode_select = driver.wait_for_xpath('//*[@id="root_observation_type"]')
+    driver.scroll_to_element(mode_select)
+    ActionChains(driver).move_to_element(mode_select).pause(1).click().perform()
+
+    mix_n_match_option = driver.wait_for_xpath('''//li[@data-value="Mix 'n Match"]''')
+    driver.scroll_to_element_and_click(mix_n_match_option)
+
+    u_band_option = driver.wait_for_xpath('//input[@id="root_observation_choices_0"]')
+
+    driver.scroll_to_element_and_click(u_band_option)
+
+    ifu_option = driver.wait_for_xpath('//input[@id="root_observation_choices_4"]')
+
+    driver.scroll_to_element_and_click(ifu_option)
+
+    driver.scroll_to_element_and_click(submit_button)
+
+    driver.wait_for_xpath(
+        f'//table[contains(@data-testid, "followupRequestTable")]//td[contains(., "Mix \'n Match")]'
+    )
+    driver.wait_for_xpath(
+        f'''//table[contains(@data-testid, "followupRequestTable")]//td[contains(., "u,IFU")]'''
+    )
+    driver.wait_for_xpath(
+        f'''//table[contains(@data-testid, "followupRequestTable")]//td[contains(., "1")]'''
+    )
+    driver.wait_for_xpath(
+        f'''//table[contains(@data-testid, "followupRequestTable")]//td[contains(., "submitted")]'''
+    )
+
+    filename = glob.glob(
+        f'{os.path.dirname(__file__)}/../data/ZTF20abwdwoa_20200902_P60_v1.ascii'
```

Comment:
Does this have to be hardcoded?

Reply:
Yeah, the data file is a part of the repo.